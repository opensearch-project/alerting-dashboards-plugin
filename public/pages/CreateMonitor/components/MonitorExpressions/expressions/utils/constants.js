/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export const POPOVER_STYLE = { zIndex: '200' };
export const EXPRESSION_STYLE = { padding: '20px', whiteSpace: 'nowrap' };
export const Expressions = {
  THRESHOLD: 'THRESHOLD',
  WHEN: 'WHEN',
  OF_FIELD: 'OF_FIELD',
  OVER: 'OVER',
  FOR_THE_LAST: 'FOR_THE_LAST',
  WHERE: 'WHERE',
  METRICS: 'METRICS',
  GROUP_BY: 'GROUP_BY',
};
export const NUMBER_TYPES = [
  'long',
  'integer',
  'short',
  'byte',
  'double',
  'float',
  'half_float',
  'scaled_float',
];
export const UNITS_OF_TIME = [
  { value: 'm', text: 'minute(s)' },
  { value: 'h', text: 'hour(s)' },
  { value: 'd', text: 'day(s)' },
];

export const WHERE_BOOLEAN_FILTERS = [
  { text: 'Select value', value: '' },
  { text: 'True', value: true },
  { text: 'False', value: false },
];

export const OPERATORS_MAP = {
  IS: {
    text: 'is',
    value: 'is',
    dataTypes: ['number', 'text', 'keyword', 'boolean'],
  },
  IS_NOT: {
    text: 'is not',
    value: 'is_not',
    dataTypes: ['number', 'text', 'keyword', 'boolean'],
  },
  IS_NULL: {
    text: 'is null',
    value: 'is_null',
    dataTypes: ['number', 'text', 'keyword', 'boolean'],
  },
  IS_NOT_NULL: {
    text: 'is not null',
    value: 'is_not_null',
    dataTypes: ['number', 'text', 'keyword'],
  },
  IS_GREATER: {
    text: 'is greater than',
    value: 'is_greater',
    dataTypes: ['number'],
  },
  IS_GREATER_EQUAL: {
    text: 'is greater than equal',
    value: 'is_greater_equal',
    dataTypes: ['number'],
  },
  IS_LESS: {
    text: 'is less than',
    value: 'is_less',
    dataTypes: ['number'],
  },
  IS_LESS_EQUAL: {
    text: 'is less than equal',
    value: 'is_less_equal',
    dataTypes: ['number'],
  },
  STARTS_WITH: {
    text: 'starts with',
    value: 'starts_with',
    dataTypes: ['text', 'keyword'],
  },
  ENDS_WITH: {
    text: 'ends with',
    value: 'ends_with',
    dataTypes: ['text', 'keyword'],
  },
  CONTAINS: {
    text: 'contains',
    value: 'contains',
    dataTypes: ['text', 'keyword'],
  },
  DOES_NOT_CONTAINS: {
    text: 'does not contain',
    value: 'does_not_contains',
    dataTypes: ['text'],
  },
  IN_RANGE: {
    text: 'is in range',
    value: 'in_range',
    dataTypes: ['number'],
  },
  NOT_IN_RANGE: {
    text: 'is not in range',
    value: 'not_in_range',
    dataTypes: ['number'],
  },
};

export const OVER_TYPES = [{ value: 'all documents', text: 'all documents' }];

export const AGGREGATION_TYPES = [
  { value: 'avg', text: 'average()' },
  { value: 'count', text: 'count()' },
  { value: 'sum', text: 'sum()' },
  { value: 'min', text: 'min()' },
  { value: 'max', text: 'max()' },
];

export const GROUP_BY_ERROR = 'Must specify at least 1 group by expression.';
export const QUERY_TYPE_GROUP_BY_ERROR = 'Can have a maximum of 1 group by selections.';

export const QUERY_TYPE_METRIC_ERROR = 'Can have a maximum of 1 metric selections.';

export const MAX_NUM_QUERY_LEVEL_GROUP_BYS = 1;
export const MAX_NUM_BUCKET_LEVEL_GROUP_BYS = 2;
export const MAX_NUM_WHERE_EXPRESSION = 1;
export const WHERE_FILTER_ALLOWED_TYPES = ['number', 'text', 'keyword', 'boolean'];
