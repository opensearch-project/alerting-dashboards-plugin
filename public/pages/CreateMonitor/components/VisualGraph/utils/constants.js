/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

export const SIZE_RANGE = [3, 3];
export const LINE_STYLES = { strokeWidth: 1 };
export const ANNOTATION_STYLES = { strokeWidth: 1, stroke: 'red' };
export const HINT_STYLES = {
  background: '#3a3a48',
  borderRadius: '5px',
  padding: '7px 10px',
  fontSize: '12px',
  fontWeight: 'bold',
  color: '#fff',
  textAlign: 'left',
  whiteSpace: 'nowrap',
  boxShadow: '0 2px 4px rgba(0,0,0,.5)',
};

// Don't want values to reach top of graph so increase by a buffer % so we always have a yDomain with a slightly higher max
export const Y_DOMAIN_BUFFER = 1.4; // 40%

export const X_DOMAIN_BUFFER = 0.2; // 40%

// Size of circle for each point on graph
export const DEFAULT_MARK_SIZE = 3;

export const BAR_PERCENTAGE = 0.015;

export const BAR_KEY_COUNT = 3;
