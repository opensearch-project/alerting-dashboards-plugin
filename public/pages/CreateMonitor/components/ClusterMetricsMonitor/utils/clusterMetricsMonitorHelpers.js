/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import _ from 'lodash';
import { formikToClusterMetricsInput } from '../../../containers/CreateMonitor/utils/formikToMonitor';
import {
  API_TYPES,
  GET_API_TYPE_DEBUG_TEXT,
  ILLEGAL_PATH_PARAMETER_CHARACTERS,
  NO_PATH_PARAMS_PLACEHOLDER_TEXT,
  PATH_PARAMETER_ILLEGAL_CHARACTER_TEXT,
  PATH_PARAMETERS_REQUIRED_TEXT,
} from './clusterMetricsMonitorConstants';
import { FORMIK_INITIAL_VALUES } from '../../../containers/CreateMonitor/utils/constants';

export function buildClusterMetricsRequest(values) {
  return _.get(formikToClusterMetricsInput(values), 'uri');
}

export const canExecuteClusterMetricsMonitor = (uri = {}) => {
  const {
    api_type = FORMIK_INITIAL_VALUES.uri.api_type,
    path_params = FORMIK_INITIAL_VALUES.uri.path_params,
  } = uri;
  if (_.isEmpty(api_type)) return false;
  const requiresPathParams = _.isEmpty(_.get(API_TYPES, `${api_type}.paths.withoutPathParams`));
  const hasPathParams = !_.isEmpty(path_params);
  if (pathParamsContainIllegalCharacters(path_params)) return false;
  return requiresPathParams ? hasPathParams : true;
};

export const pathParamsContainIllegalCharacters = (pathParams) => {
  if (_.isEmpty(pathParams)) return false;
  const foundIllegalCharacters = ILLEGAL_PATH_PARAMETER_CHARACTERS.find((illegalCharacter) =>
    _.includes(pathParams, illegalCharacter)
  );
  return !_.isEmpty(foundIllegalCharacters);
};

export const getApiPath = (hasPathParams = false, apiType) => {
  let path = hasPathParams
    ? _.get(
        API_TYPES,
        `${apiType}.paths.withPathParams`,
        _.get(API_TYPES, `${apiType}.paths.withoutPathParams`)
      )
    : _.get(API_TYPES, `${apiType}.paths.withoutPathParams`);
  return path || FORMIK_INITIAL_VALUES.uri.path;
};

export const getApiTypesRequiringPathParams = () => {
  const apiList = [];
  _.keys(API_TYPES).forEach((api) => {
    const withoutPathParams = _.get(API_TYPES, `${api}.paths.withoutPathParams`, '');
    if (_.isEmpty(withoutPathParams))
      apiList.push({
        value: _.get(API_TYPES, `${api}.type`),
        label: _.get(API_TYPES, `${api}.label`),
      });
  });
  return apiList;
};

export const getExamplePathParams = (apiType) => {
  if (_.isEmpty(apiType)) return NO_PATH_PARAMS_PLACEHOLDER_TEXT;
  let exampleText = _.get(API_TYPES, `${apiType}.exampleText`, '');
  _.isEmpty(exampleText)
    ? (exampleText = NO_PATH_PARAMS_PLACEHOLDER_TEXT)
    : (exampleText = `e.g., ${exampleText}`);
  return exampleText;
};

export const getApiType = (uri) => {
  let apiType = '';
  // Trim '/' characters from the beginning and end of the path
  const path = _.get(uri, 'path')?.replace(/^\/+|\/+$/g, '');
  if (_.isEmpty(path)) return apiType;
  _.keys(API_TYPES).forEach((apiTypeKey) => {
    const withPathParams = _.get(API_TYPES, `${apiTypeKey}.paths.withPathParams`);
    const withoutPathParams = _.get(API_TYPES, `${apiTypeKey}.paths.withoutPathParams`);
    if (path === withPathParams || path === withoutPathParams) apiType = apiTypeKey;
  });
  if (_.isEmpty(apiType)) console.debug(GET_API_TYPE_DEBUG_TEXT);
  return apiType;
};

export const isInvalidApiPathParameter = (field, hidePathParams, pathParams, requirePathParams) => {
  if (hidePathParams) return false;
  const pathParamsTouched = _.get(field, 'touched.uri.path_params', false);
  if (pathParamsTouched) {
    if (requirePathParams && _.isEmpty(pathParams)) return true;
    return pathParamsContainIllegalCharacters(pathParams);
  } else return pathParamsTouched;
};

export const validateApiPathParameter = (field, hidePathParams, pathParams, requirePathParams) => {
  if (hidePathParams) return NO_PATH_PARAMS_PLACEHOLDER_TEXT;
  const pathParamsTouched = _.get(field, 'touched.uri.path_params', false);
  if (requirePathParams && pathParamsTouched && _.isEmpty(pathParams))
    return PATH_PARAMETERS_REQUIRED_TEXT;
  if (isInvalidApiPathParameter(field, hidePathParams, pathParams, requirePathParams))
    return PATH_PARAMETER_ILLEGAL_CHARACTER_TEXT;
};
