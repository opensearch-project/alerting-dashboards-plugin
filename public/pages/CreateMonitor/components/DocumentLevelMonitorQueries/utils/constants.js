/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { OPERATORS_MAP } from '../../MonitorExpressions/expressions/utils/constants';

export const DOC_LEVEL_INPUT_FIELD = 'doc_level_input';

/**
 * A list of the operators currently supported for defining queries through the UI.
 */
export const SUPPORTED_DOC_LEVEL_QUERY_OPERATORS = [
  OPERATORS_MAP.IS.value,
  OPERATORS_MAP.IS_NOT.value,
  OPERATORS_MAP.IS_GREATER.value,
  OPERATORS_MAP.IS_GREATER_EQUAL.value,
  OPERATORS_MAP.IS_LESS.value,
  OPERATORS_MAP.IS_LESS_EQUAL.value,
];

/**
 * Initial implementation of document level monitors used these feature-specific operator values.
 * Refactored document level monitors assets to reuse the OPERATORS_MAP to reduce duplicate code
 * as new query types become supported.
 */
export const LEGACY_QUERY_OPERATORS = {
  IS: { text: 'is', value: '==' },
  IS_NOT: { text: 'is not', value: '!=' },
};

/**
 * These patterns delineate the field being queries from the query value
 * when using query string query syntax.
 */
export const QUERY_STRING_QUERY_OPERATORS = {
  [OPERATORS_MAP.IS_GREATER.value]: ':>',
  [OPERATORS_MAP.IS_GREATER_EQUAL.value]: ':>=',
  [OPERATORS_MAP.IS_LESS.value]: ':<',
  [OPERATORS_MAP.IS_LESS_EQUAL.value]: ':<=',
};

/**
 * Similar to OPERATORS_QUERY_MAP, this const contains the query patterns
 * for the SUPPORTED_DOC_LEVEL_QUERY_OPERATORS.
 */
export const DOC_LEVEL_QUERY_MAP = {
  [LEGACY_QUERY_OPERATORS.IS.value]: {
    query: ({ field, query }) => `${field}:\"${query}\"`,
  },
  [LEGACY_QUERY_OPERATORS.IS_NOT.value]: {
    query: ({ field, query }) => `NOT (${field}:\"${query}\")`,
  },
  [OPERATORS_MAP.IS.value]: {
    query: ({ field, query }) => `${field}:\"${query}\"`,
  },
  [OPERATORS_MAP.IS_NOT.value]: {
    query: ({ field, query }) => `NOT (${field}:\"${query}\")`,
  },
  [OPERATORS_MAP.IS_GREATER.value]: {
    query: ({ field, query }) =>
      `${field}${QUERY_STRING_QUERY_OPERATORS[OPERATORS_MAP.IS_GREATER.value]}${query}`,
  },
  [OPERATORS_MAP.IS_GREATER_EQUAL.value]: {
    query: ({ field, query }) =>
      `${field}${QUERY_STRING_QUERY_OPERATORS[OPERATORS_MAP.IS_GREATER_EQUAL.value]}${query}`,
  },
  [OPERATORS_MAP.IS_LESS.value]: {
    query: ({ field, query }) =>
      `${field}${QUERY_STRING_QUERY_OPERATORS[OPERATORS_MAP.IS_LESS.value]}${query}`,
  },
  [OPERATORS_MAP.IS_LESS_EQUAL.value]: {
    query: ({ field, query }) =>
      `${field}${QUERY_STRING_QUERY_OPERATORS[OPERATORS_MAP.IS_LESS_EQUAL.value]}${query}`,
  },
};
