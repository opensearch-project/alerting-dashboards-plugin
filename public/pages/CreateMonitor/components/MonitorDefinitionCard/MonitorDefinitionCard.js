/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import _ from 'lodash';
import { EuiFlexGroup, EuiFlexItem, EuiLink, EuiText } from '@elastic/eui';
import FormikCheckableCard from '../../../../components/FormControls/FormikCheckableCard/FormikCheckableCard';
import { OS_AD_PLUGIN, MONITOR_TYPE, SEARCH_TYPE } from '../../../../utils/constants';
import { MONITOR_TYPE_CARD_WIDTH } from '../MonitorType/MonitorType';
import { URL } from '../../../../../utils/constants';

const MONITOR_DEFINITION_CARD_WIDTH =
  (MONITOR_TYPE_CARD_WIDTH * _.keys(MONITOR_TYPE).length) / _.keys(SEARCH_TYPE).length;

const onChangeDefinition = (e, form) => {
  const type = e.target.value;
  form.setFieldValue('searchType', type, false);
};

const MonitorDefinitionCard = ({ values, plugins }) => {
  const hasADPlugin = plugins.indexOf(OS_AD_PLUGIN) !== -1;
  const isBucketLevelMonitor = values.monitor_type === MONITOR_TYPE.BUCKET_LEVEL;

  return (
    <div>
      <EuiText size={'xs'} style={{ paddingBottom: '0px', marginBottom: '0px' }}>
        <h4>Monitor defining method</h4>
      </EuiText>
      <EuiText color={'subdued'} size={'xs'}>
        Specify the way you want to define your query and triggers.{' '}
        <EuiLink external href={URL.CREATE_MONITOR_DOCUMENTATION} target="_blank">
          Learn more
        </EuiLink>
      </EuiText>
      <EuiFlexGroup
        alignItems={'flexStart'}
        gutterSize={'s'}
        style={{ paddingTop: '0px', marginTop: '0px' }}
      >
        <EuiFlexItem grow={false} style={{ width: `${MONITOR_DEFINITION_CARD_WIDTH}px` }}>
          <FormikCheckableCard
            name="searchTypeGraph"
            formRow
            inputProps={{
              id: 'visualEditorRadioCard',
              label: 'Visual editor',
              checked: values.searchType === SEARCH_TYPE.GRAPH,
              value: SEARCH_TYPE.GRAPH,
              onChange: (e, field, form) => {
                onChangeDefinition(e, form);
              },
              'data-test-subj': 'visualEditorRadioCard',
            }}
          />
        </EuiFlexItem>
        <EuiFlexItem grow={false} style={{ width: `${MONITOR_DEFINITION_CARD_WIDTH}px` }}>
          <FormikCheckableCard
            name="searchTypeQuery"
            formRow
            inputProps={{
              id: 'extractionQueryEditorRadioCard',
              label: 'Extraction query editor',
              checked: values.searchType === SEARCH_TYPE.QUERY,
              value: SEARCH_TYPE.QUERY,
              onChange: (e, field, form) => {
                onChangeDefinition(e, form);
              },
              'data-test-subj': 'extractionQueryEditorRadioCard',
            }}
          />
        </EuiFlexItem>
        {/*// Only show the anomaly detector option when anomaly detection plugin is present, but not for bucket-level monitors.*/}
        {hasADPlugin && !isBucketLevelMonitor && (
          <EuiFlexItem grow={false} style={{ width: `${MONITOR_DEFINITION_CARD_WIDTH}px` }}>
            <FormikCheckableCard
              name="searchTypeAD"
              inputProps={{
                id: 'anomalyDetectorRadioCard',
                label: 'Anomaly detector',
                checked: values.searchType === SEARCH_TYPE.AD,
                value: SEARCH_TYPE.AD,
                onChange: (e, field, form) => {
                  onChangeDefinition(e, form);
                },
                'data-test-subj': 'anomalyDetectorRadioCard',
              }}
            />
          </EuiFlexItem>
        )}
      </EuiFlexGroup>
    </div>
  );
};

export default MonitorDefinitionCard;
