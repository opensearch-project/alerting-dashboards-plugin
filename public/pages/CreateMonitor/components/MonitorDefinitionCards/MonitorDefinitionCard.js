/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import _ from 'lodash';
import { EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
import FormikCheckableCard from '../../../../components/FormControls/FormikCheckableCard/FormikCheckableCard';
import { OS_AD_PLUGIN, MONITOR_TYPE, SEARCH_TYPE } from '../../../../utils/constants';
import { MONITOR_TYPE_CARD_WIDTH } from '../MonitorType/MonitorType';

const MONITOR_DEFINITION_CARD_WIDTH =
  (MONITOR_TYPE_CARD_WIDTH * _.keys(MONITOR_TYPE).length) / _.keys(SEARCH_TYPE).length;

const onChangeDefinition = (e, form) => {
  const type = e.target.value;
  form.setFieldValue('searchType', type, false);
};

const MonitorDefinitionCard = ({ values, resetResponse, plugins }) => {
  const hasADPlugin = plugins.indexOf(OS_AD_PLUGIN) !== -1;
  const isBucketLevelMonitor = values.monitor_type === MONITOR_TYPE.BUCKET_LEVEL;

  return (
    <div>
      <EuiFlexGroup alignItems={'flexEnd'} gutterSize={'s'}>
        <EuiFlexItem grow={false} style={{ width: `${MONITOR_DEFINITION_CARD_WIDTH}px` }}>
          <FormikCheckableCard
            name="searchTypeGraph"
            formRow
            rowProps={{ label: 'Choose a monitor defining method' }}
            inputProps={{
              id: 'visualEditorRadioCard',
              label: 'Visual editor',
              checked: values.searchType === SEARCH_TYPE.GRAPH,
              value: SEARCH_TYPE.GRAPH,
              onChange: (e, field, form) => {
                onChangeDefinition(e, form);
              },
            }}
          />
        </EuiFlexItem>
        <EuiFlexItem grow={false} style={{ width: `${MONITOR_DEFINITION_CARD_WIDTH}px` }}>
          <EuiSpacer />
          <FormikCheckableCard
            name="searchTypeQuery"
            formRow
            inputProps={{
              id: 'extractionQueryEditorRadioCard',
              label: 'Extraction query editor',
              checked: values.searchType === SEARCH_TYPE.QUERY,
              value: SEARCH_TYPE.QUERY,
              onChange: (e, field, form) => {
                onChangeDefinition(e, form);
              },
            }}
          />
        </EuiFlexItem>
        {/*// Only show the anomaly detector option when anomaly detection plugin is present, but not for bucket-level monitors.*/}
        {hasADPlugin && !isBucketLevelMonitor && (
          <EuiFlexItem grow={false} style={{ width: `${MONITOR_DEFINITION_CARD_WIDTH}px` }}>
            <EuiSpacer />
            <FormikCheckableCard
              name="searchTypeAD"
              inputProps={{
                id: 'anomalyDetectorRadioCard',
                label: 'Anomaly detector',
                checked: values.searchType === SEARCH_TYPE.AD,
                value: SEARCH_TYPE.AD,
                onChange: (e, field, form) => {
                  onChangeDefinition(e, form);
                },
              }}
            />
          </EuiFlexItem>
        )}
      </EuiFlexGroup>
    </div>
  );
};

export default MonitorDefinitionCard;
