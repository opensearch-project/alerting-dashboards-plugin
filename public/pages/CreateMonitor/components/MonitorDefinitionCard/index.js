/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import MonitorDefinitionCard from './MonitorDefinitionCard';

export default MonitorDefinitionCard;
