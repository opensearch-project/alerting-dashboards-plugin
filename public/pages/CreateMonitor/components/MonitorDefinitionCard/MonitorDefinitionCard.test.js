/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';

import MonitorDefinitionCard from './MonitorDefinitionCard';
import { Formik } from 'formik';
import { FORMIK_INITIAL_VALUES } from '../../containers/CreateMonitor/utils/constants';
import { OS_AD_PLUGIN } from '../../../../utils/constants';

describe('MonitorDefinitionCard', () => {
  test('renders without AD plugin', () => {
    const component = (
      <Formik initialValues={FORMIK_INITIAL_VALUES}>
        {() => <MonitorDefinitionCard values={{}} plugins={[]} />}
      </Formik>
    );

    expect(render(component)).toMatchSnapshot();
  });
  test('renders without AD plugin', () => {
    const component = (
      <Formik initialValues={FORMIK_INITIAL_VALUES}>
        {() => <MonitorDefinitionCard values={{}} plugins={[OS_AD_PLUGIN]} />}
      </Formik>
    );

    expect(render(component)).toMatchSnapshot();
  });
});
