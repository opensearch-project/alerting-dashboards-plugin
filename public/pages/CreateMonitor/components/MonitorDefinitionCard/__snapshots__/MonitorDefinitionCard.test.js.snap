// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonitorDefinitionCard renders without AD plugin 1`] = `
<div>
  <div
    class="euiText euiText--extraSmall"
    style="padding-bottom:0px;margin-bottom:0px"
  >
    <h4>
      Monitor defining method
    </h4>
  </div>
  <div
    class="euiText euiText--extraSmall"
  >
    <div
      class="euiTextColor euiTextColor--subdued"
    >
      Specify the way you want to define your query and triggers. 
      <a
        class="euiLink euiLink--primary"
        href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#create-monitors"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn more
        <div>
          EuiIconMock
        </div>
        <span
          class="euiScreenReaderOnly"
        >
          (opens in a new tab or window)
        </span>
      </a>
    </div>
  </div>
  <div
    class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsFlexStart euiFlexGroup--directionRow euiFlexGroup--responsive"
    style="padding-top:0px;margin-top:0px"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
      style="width:275px"
    >
      <div
        class="euiFormRow"
        id="searchTypeGraph-form-row-row"
      >
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--hasBorder euiPanel--flexGrowZero euiSplitPanel euiSplitPanel--row euiCheckableCard"
          >
            <div
              class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero euiPanel--isClickable euiSplitPanel__inner"
            >
              <div
                class="euiRadio euiRadio--noLabel"
                data-test-subj="visualEditorRadioCard"
              >
                <input
                  class="euiRadio__input"
                  id="visualEditorRadioCard"
                  name="searchTypeGraph"
                  type="radio"
                  value="graph"
                />
                <div
                  class="euiRadio__circle"
                />
              </div>
            </div>
            <div
              class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiSplitPanel__inner"
            >
              <label
                class="euiCheckableCard__label"
                for="visualEditorRadioCard"
              >
                Visual editor
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
      style="width:275px"
    >
      <div
        class="euiFormRow"
        id="searchTypeQuery-form-row-row"
      >
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--hasBorder euiPanel--flexGrowZero euiSplitPanel euiSplitPanel--row euiCheckableCard"
          >
            <div
              class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero euiPanel--isClickable euiSplitPanel__inner"
            >
              <div
                class="euiRadio euiRadio--noLabel"
                data-test-subj="extractionQueryEditorRadioCard"
              >
                <input
                  class="euiRadio__input"
                  id="extractionQueryEditorRadioCard"
                  name="searchTypeQuery"
                  type="radio"
                  value="query"
                />
                <div
                  class="euiRadio__circle"
                />
              </div>
            </div>
            <div
              class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiSplitPanel__inner"
            >
              <label
                class="euiCheckableCard__label"
                for="extractionQueryEditorRadioCard"
              >
                Extraction query editor
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MonitorDefinitionCard renders without AD plugin 2`] = `
<div>
  <div
    class="euiText euiText--extraSmall"
    style="padding-bottom:0px;margin-bottom:0px"
  >
    <h4>
      Monitor defining method
    </h4>
  </div>
  <div
    class="euiText euiText--extraSmall"
  >
    <div
      class="euiTextColor euiTextColor--subdued"
    >
      Specify the way you want to define your query and triggers. 
      <a
        class="euiLink euiLink--primary"
        href="https://opensearch.org/docs/monitoring-plugins/alerting/monitors/#create-monitors"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn more
        <div>
          EuiIconMock
        </div>
        <span
          class="euiScreenReaderOnly"
        >
          (opens in a new tab or window)
        </span>
      </a>
    </div>
  </div>
  <div
    class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsFlexStart euiFlexGroup--directionRow euiFlexGroup--responsive"
    style="padding-top:0px;margin-top:0px"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
      style="width:275px"
    >
      <div
        class="euiFormRow"
        id="searchTypeGraph-form-row-row"
      >
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--hasBorder euiPanel--flexGrowZero euiSplitPanel euiSplitPanel--row euiCheckableCard"
          >
            <div
              class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero euiPanel--isClickable euiSplitPanel__inner"
            >
              <div
                class="euiRadio euiRadio--noLabel"
                data-test-subj="visualEditorRadioCard"
              >
                <input
                  class="euiRadio__input"
                  id="visualEditorRadioCard"
                  name="searchTypeGraph"
                  type="radio"
                  value="graph"
                />
                <div
                  class="euiRadio__circle"
                />
              </div>
            </div>
            <div
              class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiSplitPanel__inner"
            >
              <label
                class="euiCheckableCard__label"
                for="visualEditorRadioCard"
              >
                Visual editor
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
      style="width:275px"
    >
      <div
        class="euiFormRow"
        id="searchTypeQuery-form-row-row"
      >
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--hasBorder euiPanel--flexGrowZero euiSplitPanel euiSplitPanel--row euiCheckableCard"
          >
            <div
              class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero euiPanel--isClickable euiSplitPanel__inner"
            >
              <div
                class="euiRadio euiRadio--noLabel"
                data-test-subj="extractionQueryEditorRadioCard"
              >
                <input
                  class="euiRadio__input"
                  id="extractionQueryEditorRadioCard"
                  name="searchTypeQuery"
                  type="radio"
                  value="query"
                />
                <div
                  class="euiRadio__circle"
                />
              </div>
            </div>
            <div
              class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiSplitPanel__inner"
            >
              <label
                class="euiCheckableCard__label"
                for="extractionQueryEditorRadioCard"
              >
                Extraction query editor
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
